### PR TITLE
fix: manage to do some more strict related items

### PR DESCRIPTION
**Pull request by rebelchris**
## Changes

Some more work around strict mode for webapp (so far)

## Events

Did you introduce any new tracking events?

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

### Preview domain
https://chore-strict-v2.preview.app.daily.dev

### CODE REVIEW

File: packages/shared/src/contexts/AuthContext.tsx [hunk 1]
```diff
@@ -99,7 +99,7 @@ const logout = async (reason: string): Promise<void> => {
 };
 
 export type AuthContextProviderProps = {
-  user: LoggedUser | AnonymousUser | undefined;
+  user?: LoggedUser | AnonymousUser;
```
Comment (rebelchris) [inline packages/shared/src/contexts/AuthContext.tsx:102]: I personally prefer this, but let me know if people liked the otherside better

Comment (omBratteng) [reply packages/shared/src/contexts/AuthContext.tsx:102]: Same, there's a reason why `?:` exists 😸

**Review by sshanzel — APPROVED** [review]
Looks amazing! We'll get there eventually 🥳

File: packages/shared/src/contexts/AuthContext.tsx (hunk 1 above)
Comment (capJavert) [reply packages/shared/src/contexts/AuthContext.tsx:102]: +1

File: packages/webapp/pages/[userId]/posts.tsx [hunk 2]
```diff
@@ -42,7 +42,7 @@ const ProfilePostsPage = ({ user }: ProfileLayoutProps): ReactElement => {
       />
     ) : (
       <ProfileEmptyScreen
-        title={`${user.name} hasn't posted yet`}
+        title={`${user?.name ?? 'User'} hasn't posted yet`}
```
Comment (capJavert) [inline packages/webapp/pages/[userId]/posts.tsx:45]: Just had similar type issue on API side 😄

File: packages/webapp/pages/account/profile.tsx [hunk 3]
```diff
@@ -94,7 +94,7 @@ const AccountProfilePage = (): ReactElement => {
       }),
     {
       onSuccess: async (res) => {
-        await updateUser({ ...user, cover: res.user.cover });
+        await updateUser({ ...user, cover: res.user.cover } as LoggedUser);
```
Comment (capJavert) [inline packages/webapp/pages/account/profile.tsx:97]: it does not infer?  ah probably because `user` can be `undefined`

**Review by capJavert — APPROVED** [review]

File: packages/webapp/pages/account/profile.tsx (hunk 3 above)
Comment (rebelchris) [reply packages/webapp/pages/account/profile.tsx:97]: It's the difference between `LoggedUser` and `AnnonymousUser` one has mandatory cover, other doesn't basically.

**Pull request merged**